src/tasks/local_verify.ts

```diff
@@ -49,8 +49,45 @@ task("local-verify", "Verifies that the local deployment files correspond to the
             const deployment = await hre.deployments.get(contract);
             const onChainCode = await hre.ethers.provider.getCode(deployment.address);
             const onchainBytecodeHash = hre.ethers.keccak256(onChainCode);
-            // TODO: compile contract in realtime and compare the compiled bytecode with onchain bytecode
-            const localBytecodeHash = hre.ethers.keccak256(deployment.deployedBytecode!);
```

Comment:
I am fairly sure that `hre.deployments.get(contract)` already compiles the contract locally with the configured Hardhat settings, so I don't think the Solidity compiler needs to be invoked directly here.

I'm a bit surprised at the comment TBH.